### PR TITLE
feat(docs): add cuhacking google calendar

### PR DESCRIPTION
**Pull request by AquaShotRyan**
Screenshot 2024-08-31 154631
Screenshot 2024-08-31 154636
Screenshot 2024-08-31 154640
Screenshot 2024-08-31 154645
Screenshot 2024-08-31 154649
Screenshot 2024-08-31 154654

## Summary by Sourcery

Add a Google Calendar integration to the documentation site using FullCalendar, allowing users to view events in different formats. Update the build configuration to include necessary dependencies and embed the calendar in the documentation index page.

New Features:
- Integrate a Google Calendar into the documentation site using the FullCalendar library, allowing users to view events in various formats such as week, month, and list views.

Enhancements:
- Add a new Calendar component to the documentation site, which utilizes environment variables for Google Calendar API integration.

Build:
- Update pnpm-lock.yaml to include new dependencies for FullCalendar plugins and Preact.

Documentation:
- Embed the new Calendar component into the documentation index page, providing users with a visual calendar interface.

## Summary by CodeRabbit

- **New Features**
	- Introduced a calendar component that allows users to view and manage events from Google Calendar.
	- Added customizable calendar views, including day grid, time grid, and list view.

- **Style**
	- Implemented new CSS styles for tables within prose elements to enhance visual presentation.

- **Documentation**
	- Updated documentation to include the new calendar component, providing users with integration details for Google Calendar.
	- Added instructions for new environment variables related to Google Calendar in the `.env.example` file.

- **Chores**
	- Added necessary dependencies for FullCalendar functionality to the project.
	- Updated workflows to include environment variables for Google Calendar integration.

### CODE REVIEW

**Comment by MFarabi619** [comment]
For dependencies you can do:
- `chore(deps): add @fullcalendar/daygrid`
- `chore(deps): add @fullcalendar/...`
- `chore(deps): add ....`

File: apps/docs/src/components/Calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import googleCalendarPlugin from '@fullcalendar/google-calendar'
+
+// cuHacking calendar
+const GOOGLE_CALENDAR_ID = 'fcdba3f354d4e01552e2495d743105bd9efce4e1076cda3c1ec1454d0f5faa73@group.calendar.google.com'
+
+// temporary, add as an env variable later
+// google calendar api key
+const API_KEY = ''
+
+export default function Calendar() {
```
Comment (MFarabi619) [inline apps/docs/src/components/Calendar.tsx:16]: Best practice is to [avoid default exports](https://dev.to/phuocng/avoid-using-default-exports-a1c)

File: apps/docs/src/components/Calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      }}
+      initialView="timeGridWeek"
+      headerToolbar={{
+        left: 'today prev,next',
+        center: 'title',
+        right: 'timeGridWeek listWeek dayGridMonth',
+      }}
+      views={{
+        timeGridWeek: {
+          slotMinTime: '06:00:00',
```
Comment (MFarabi619) [inline apps/docs/src/components/Calendar.tsx:32]: Nice

File: apps/docs/src/components/Calendar.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+'use client'
```
Comment (MFarabi619) [inline apps/docs/src/components/Calendar.tsx:1]: Is it possible to remove this and render the component server-side? Then we can move all of the dependencies to dev dependencies, and the bundle size will be much smaller for the client. 

What would be the other tradeoffs? Does this calendar update real-time?

**Review by MFarabi619 — CHANGES_REQUESTED** [review]

File: apps/docs/src/components/Calendar.tsx (hunk 3 above)
Comment (AquaShotRyan) [reply apps/docs/src/components/Calendar.tsx:1]: I removed it and it gave me this error:
image
I did some research online and there doesn't seem to be a way to render it server-side, at least as a React component. I think it needs to be client side in order to be interactive.

**Comment by AquaShotRyan** [comment]
@sourcery-ai review

**Comment by AquaShotRyan** [comment]
@sourcery-ai review

File: apps/docs/src/components/Calendar.tsx (hunk 3 above)
Comment (MFarabi619) [reply apps/docs/src/components/Calendar.tsx:1]: Okay sounds good

File: apps/docs/src/components/Calendar.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      views={{
+        timeGridWeek: {
+          slotMinTime: '06:00:00',
+          slotMaxTime: '24:00:00',
+          scrollTime: '17:00:00',
+        },
+        dayGridMonth: {},
+        listWeek: {},
+      }}
+      eventColor="red"
```
[the remark this answers is not in the document]
Comment (MFarabi619) [reply apps/docs/src/components/Calendar.tsx]: Amazing PR reviews @sourcery

**Comment by AquaShotRyan** [comment]
In the process of learning of split commits (i.e. split the commit for installing dependencies) and adding more styles to the calendar.

**Comment by MFarabi619** [comment]
Let's get this in, and use our `.env`library

**Comment by MFarabi619** [comment]
Closing as stale

**Pull request closed without merging**